core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/AddOperationsToChain.java

```diff
@@ -0,0 +1,234 @@
+     * @param opChain the {@link OperationChain} being executed.
+     * @param user    the {@link User} executing the operation chain
+     */
+    @Override
+    public void preExecute(final OperationChain<?> opChain, final User user) {
+        final OperationChain<?> newOpChain = new OperationChain<>();
+
+        boolean hasAuth = false;
+        if (!authorisedOps.isEmpty() && !user.getOpAuths().isEmpty()) {
+            for (final String auth : authorisedOps.keySet()) {
+                if (user.getOpAuths().contains(auth)) {
+                    newOpChain.getOperations().addAll(addOperationsToChain(opChain, authorisedOps.get(auth)).getOperations());
+                    hasAuth = true;
+                    break;
+                }
+            }
+        }
+
+        if (!hasAuth) {
+            newOpChain.getOperations().addAll(addOperationsToChain(opChain, defaultOperations).getOperations());
+        }
+
+        opChain.getOperations().clear();
+        opChain.getOperations().addAll(newOpChain.getOperations());
+    }
+
+    @Override
+    public <T> T postExecute(final T result,
+                             final OperationChain<?> opChain, final User user) {
+        return result;
+    }
+
+    private static AddOperationsToChain fromJson(final byte[] json) {
+        try {
+            return JSON_SERIALISER.deserialise(json, AddOperationsToChain.class);
+        } catch (final SerialisationException e) {
+            throw new SchemaException("Failed to load element definitions from bytes", e);
+        }
+    }
+
+    private static AddOperationsToChain fromJson(final InputStream inputStream) {
+        try {
+            return JSON_SERIALISER.deserialise(inputStream, AddOperationsToChain.class);
+        } catch (final SerialisationException e) {
+            throw new SchemaException("Failed to load element definitions from bytes", e);
+        }
+    }
+
+    private void setOperations(final AddOperationsToChain addOperations) {
+        if (addOperations.getAuthorisedOps() != null) {
+            this.setAuthorisedOps(addOperations.getAuthorisedOps());
+        }
+        if (addOperations.getDefaultOperations().getStart() != null) {
+            this.getDefaultOperations().setStart(addOperations.getDefaultOperations().getStart());
+        }
+        if (addOperations.getDefaultOperations().getEnd() != null) {
+            this.getDefaultOperations().setEnd(addOperations.getDefaultOperations().getEnd());
+        }
+        if (addOperations.getDefaultOperations().getBefore() != null) {
+            this.getDefaultOperations().setBefore(addOperations.getDefaultOperations().getBefore());
+        }
+        if (addOperations.getDefaultOperations().getAfter() != null) {
+            this.getDefaultOperations().setAfter(addOperations.getDefaultOperations().getAfter());
+        }
+    }
+
+    private OperationChain<?> addOperationsToChain(final OperationChain<?> opChain, final AdditionalOperations additionalOperations) {
```

Comment:
newOpChain could just be a List of Operation and then you could change the return type to be List of Operation.